File: 2017/ViewAnimations/ClassicAnimation/app/src/main/java/me/mehdi/classicanimation/MainActivity.java
package me.mehdi.classicanimation;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.AnimationSet;
import android.view.animation.RotateAnimation;
import android.view.animation.ScaleAnimation;
import android.view.animation.TranslateAnimation;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    Button hello;
    TranslateAnimation translate;
    RotateAnimation rotate;
    ScaleAnimation scale;
    AnimationSet animationSet;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        hello = findViewById(R.id.hello);

        translate = new TranslateAnimation(0, 200, 0, 200 );
        rotate = new RotateAnimation(0, 360);
        scale = new ScaleAnimation(0, 3, 0, 3);

        animationSet = new AnimationSet(this, null);
        animationSet.addAnimation(translate);
        animationSet.addAnimation(rotate);
        animationSet.addAnimation(scale);
        animationSet.setDuration(3000);



        hello.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                v.startAnimation(animationSet);
            }
        });
    }
}

File: 2022/ViewAnimation/app/src/main/java/com/example/viewanimation/MainActivity.kt
package com.example.viewanimation

import android.content.Intent
import android.graphics.drawable.Animatable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.animation.*
import com.example.viewanimation.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)



        binding = ActivityMainBinding.inflate(layoutInflater)


        setContentView(binding.root)


        title = "MainActivity"
        binding.start.setOnClickListener {button->
            val myResAnimSet = AnimationUtils.loadAnimation(this, R.anim.my_anim)

            myResAnimSet.setAnimationListener(object: Animation.AnimationListener{
                override fun onAnimationStart(p0: Animation?) {
                    Log.d("MainActivity", "onAnimationStart: ")
                }

                override fun onAnimationEnd(p0: Animation?) {
                    Log.d("MainActivity", "onAnimationEnd: button animation finished")
                    val intent = Intent(applicationContext, SecondActivity::class.java)
                    startActivity(intent)
                    overridePendingTransition(R.anim.slide_right_in, R.anim.slide_left_out)
                }

                override fun onAnimationRepeat(p0: Animation?) {
                    Log.d("MainActivity", "onAnimationRepeat: ")
                }

            })
            button.startAnimation(myResAnimSet)


            binding.view.setBackgroundResource(R.drawable.my_frame_anim)
            val bgAnim = binding.view.background
            if (bgAnim is Animatable){
                bgAnim.start()
            }
//            val myAnimationSet = AnimationSet(true)
//            myAnimationSet.duration = 3000
//            val translate = TranslateAnimation(
//                0f,
//            300f,
//                 0f,
//                 300f
//            )
//
//            val rotate = RotateAnimation(0f, 360f)
//            val alpha = AlphaAnimation(1f, 0f)
//            val scale = ScaleAnimation(
//                1f, 3f,
//            1f, 3f)
//
//
//            myAnimationSet.apply{
//                addAnimation(translate)
//                addAnimation(rotate)
//                addAnimation(alpha)
//                addAnimation(scale)
//            }
//
//            button.startAnimation(myAnimationSet)

        }


        binding.startActivity.setOnClickListener {
            val intent = Intent(this, SecondActivity::class.java)
            startActivity(intent)
            overridePendingTransition(R.anim.slide_right_in, R.anim.slide_left_out)

        }


    }
}